lib/kafka/offset_manager.rb

```diff
@@ -111,17 +116,51 @@ def seconds_since_last_commit
       Time.now - @last_commit
     end
 
+    def seconds_since_last_recommit
+      Time.now - @last_recommit
+    end
+
     def committed_offset_for(topic, partition)
       @committed_offsets ||= @group.fetch_offsets
       @committed_offsets.offset_for(topic, partition)
     end
 
+    def offsets_to_commit(recommit = false)
+      if recommit && !@committed_offsets.nil?
```

Comment:
While looking at this again, I decided that it's possible for the recommit to be called when `@committed_offsets` has not been repopulated. This is because `OffsetManager#next_offset_for` is called from `Consumer#fetch_batches`, but `#commit_offsets_if_necessary` may be called after each batch. So I've updated the `OffsetManager` to fetch offsets if they're missing for recommit. This meant that a couple of calls to explicitly fetch offsets could be removed from the specs.